flyteplugins/go/tasks/plugins/k8s/pod/plugin.go

```diff
@@ -172,7 +172,7 @@ func (plugin) GetTaskPhaseWithLogs(ctx context.Context, pluginContext k8s.Plugin
 	}
 
 	taskExecID := pluginContext.TaskExecutionMetadata().GetTaskExecutionID()
-	if pod.Status.Phase != v1.PodPending && pod.Status.Phase != v1.PodUnknown {
+	if pod.Status.Phase != v1.PodUnknown {
```

Comment:
Now, we do want to generate task logs while we are pending.

Reply:
Its not a big problem but I think this change might be the reason that we get quite a lot of spam in the flytepropeller logs like 
```
{"json":{"exec_id":"fybmhjrhjorydc","node":"n0/dn1","ns":"<namespace>","res_ver":"3487679492","routine":"worker-154","tasktype":"python-task","wf":"<workflow>"},"level":"error","msg":"containerStatus IndexOutOfBound, requested [0], but total containerStatuses [0] in pod phase [Pending]","ts":"2024-10-28T13:56:14Z"}
```
This comes from [this line](https://github.com/flyteorg/flyte/blob/13b3d82c94fb61047a49aa2093a2b17fb6a7c792/flyteplugins/go/tasks/logs/logging_utils.go#L34), so I wonder if we should adjust the logging a bit in response to this change.

Reply:
Thanks for bringing it to my attention that this change causes excessive logs!

Before this PR far, `GetLogsForContainerInPod` wasn't called if `pod.Status.Phase == v1.PodPending`. If in a later pod phase there still weren't any container statuses, this was unexpected, hence error log level.

Since we now optionally generate task logs already in the pending phase, if I'm not misunderstanding something, it's *not* unexpected that there might not be any container statuses at this point 🤔 

I propose to [here](https://github.com/flyteorg/flyte/blob/13b3d82c94fb61047a49aa2093a2b17fb6a7c792/flyteplugins/go/tasks/logs/logging_utils.go#L34) set the log level to debug if the pod is in pending phase and keep it at error level otherwise.

Does this make sense to you @Tom-Newton ?

Reply:
That makes sense to me 👍

Reply:
I'll make a PR 👍

Reply:
https://github.com/flyteorg/flyte/pull/5945